src/sqlfluff/rules/L059.py

```diff
@@ -83,6 +83,14 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
             context_policy = "quoted_identifier"
             identifier_contents = context.segment.raw[1:-1]
 
+        # Ignore "password_auth" type to allow quotes around passwords within
```

Comment:
I’m not loving this hard coding to be honest.

At the very least I think we should have this as an `_ignore_types` list so it’s easier to add new types (as we surely will need to in future), rather than really hardcoded within the rule. See here for a similar example:

https://github.com/sqlfluff/sqlfluff/blob/7a204ddccc66cc37c1b23199668600609459e744/src/sqlfluff/rules/L054.py#L87

Reply:
Yes, this is a way better approach. Added this.